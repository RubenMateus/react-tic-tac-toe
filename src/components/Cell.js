import React from 'react';
import { css } from 'emotion';
import PropTypes from 'prop-types';

const Cell = ({ value, onClick }) => (
  <div
    className={css`
      background-color: #fff;
      width: 100px;
      height: 100px;
    `}
  >
    <button
      type="button"
      onClick={onClick}
      className={css`
        width: 100%;
        height: 100%;
      `}
    >
      {value}
    </button>
  </div>
);

Cell.propTypes = {
  value: PropTypes.string,
  onClick: PropTypes.func.isRequired,
};

export default Cell;
