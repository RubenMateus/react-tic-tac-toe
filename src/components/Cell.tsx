import { css } from "@emotion/css";

const Cell = ({ value, onClick }) => (
  <div
    className={css`
      background-color: #fff;
      width: 100px;
      height: 100px;
    `}
  >
    <button
      type="button"
      onClick={onClick}
      className={css`
        width: 100%;
        height: 100%;
      `}
    >
      {value}
    </button>
  </div>
);

export default Cell;
