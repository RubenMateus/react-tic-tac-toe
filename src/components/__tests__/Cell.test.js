import React from 'react';
import { render } from '@testing-library/react';
import Cell from '../Cell';

describe('<Cell />', () => {
  test('renders correctly', () => {
    const { asFragment } = render(<Cell onClick={jest.fn()} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
