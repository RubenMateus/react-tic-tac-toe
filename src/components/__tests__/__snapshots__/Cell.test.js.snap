// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cell /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="css-ppfkvi"
  >
    <button
      class="css-113hvi"
      type="button"
    />
  </div>
</DocumentFragment>
`;
